tests/test_model.py

```diff
@@ -93,8 +93,35 @@ def test_setting_getting_channel_width(test_DeltaModel):
     test_DeltaModel.channel_flow_depth = 2
     assert test_DeltaModel.channel_flow_depth == 2
 
-    
+
 def test_setting_getting_channel_width(test_DeltaModel):
     assert test_DeltaModel.influx_sediment_concentration == 0.1
     test_DeltaModel.influx_sediment_concentration = 2
     assert test_DeltaModel.C0 == 0.02
+
+
+def test_make_checkpoint(tmp_path, test_DeltaModel):
+    """Test setting the checkpoint option to 'True' and saving a checkpoint."""
+    test_DeltaModel.checkpoint = True
+    test_DeltaModel.save_dt = 1
+    check_DeltaModel = test_DeltaModel
+    test_DeltaModel.update()
+    exp_path_npz = os.path.join(tmp_path / 'out_dir', 'checkpoint.npz')
+    exp_path_strateta = os.path.join(tmp_path / 'out_dir', 'strata_eta.npz')
+    exp_path_sand = os.path.join(tmp_path / 'out_dir', 'strata_sand_frac.npz')
+    assert os.path.isfile(exp_path_npz)
+    assert os.path.isfile(exp_path_strateta)
+    assert os.path.isfile(exp_path_sand)
+    # check loading checkpoint and see if it works
+    check_DeltaModel.load_checkpoint()
+    assert check_DeltaModel._time == test_DeltaModel._time
+    assert np.all(check_DeltaModel.uw == test_DeltaModel.uw)
+    assert np.all(check_DeltaModel.ux == test_DeltaModel.ux)
+    assert np.all(check_DeltaModel.uy == test_DeltaModel.uy)
+    assert np.all(check_DeltaModel.depth == test_DeltaModel.depth)
+    assert np.all(check_DeltaModel.stage == test_DeltaModel.stage)
+    assert np.all(check_DeltaModel.eta == test_DeltaModel.eta)
+    assert np.all(check_DeltaModel.strata_eta.todense() ==
+                  test_DeltaModel.strata_eta.todense())
+    assert np.all(check_DeltaModel.strata_sand_frac.todense() ==
+                  test_DeltaModel.strata_sand_frac.todense())
```

Comment:
how about another test which verifies the consistency of checkpointing? This could be in this file or the `test_consistent` file. 

Basically, set up a model with known initial conditions and run the model for 6 timesteps, and record the status of the `eta` array. Then create a new model with the same initial conditions and run for three timesteps, then checkpoint, then delete this model. Then create a third model, load the checkpoint, and run for three more timesteps. Compare the status of this model with the `eta` array grabbed from the first model. They should be identical.